Python/specialize.c

```diff
@@ -1789,8 +1789,7 @@ specialize_class_call(PyObject *callable, _Py_CODEUNIT *instr, int nargs)
         return -1;
```

Comment:
Can we handle this with `CALL_NON_PY_GENERAL`?

Reply:
Yes, but in `specialize_class_call`. 
It keeps the logic cleaner if we don't make assumptions about `specialize_class_call` failing here.